eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,50 @@
   $VsoVariable = "" # target devops output variable
 )
-$target = $TargetDirectory.ToLower().Trim("/")
-$codeOwnersLocation = Join-Path $RootDirectory -ChildPath ".github/CODEOWNERS"
-$ownedFolders = @{}
 
-if (!(Test-Path $codeOwnersLocation)) {
-  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
-  exit 1
-}
-
-$codeOwnersContent = Get-Content $codeOwnersLocation
+$EngOutputIndicator = ".*EngToolOutputProperty: (.*)"
 
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
-  }
+# Check if the retrieve-codeowners tool exsit or not.
+function installRetrieveCodeOwnersTool() {
+  dotnet tool install --tool-path $ToolPath --add-source "$DevOpsFeed" --version "$ToolVersion" "Azure.Sdk.Tools.RetrieveCodeOwners"
 }
 
-$results = $ownedFolders[$target]
-
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
-  
-  if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
+# Get the json property from tool command.
+function getCodeOwnersEntryFromCommand() {
+  return & "$ToolPath/retrieve-code-owners" --target-directory "$PathToOwners" --root-directory "$WorkingDirectory" 2>&1
+}
+function getCodeOwners() { 
+  $codeOwnerConsoleOutput = getCodeOwnersEntryFromCommand
 
-    if ($alreadyPresent) { 
-      $results += ",$alreadyPresent"
-    }
-    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$results"
+  # Failed at the command of fetching code owners.
+  if ($LASTEXITCODE -ne 0) {
+    Write-Host $codeOwnerConsoleOutput
+    return $null
   }
-
-  return $results
+  # Parsing the json property from console output.
+  if (!($codeOwnerConsoleOutput -match $EngOutputIndicator)) {
+    Write-Host "There is something wrong with tool output. It does not include eng parsing format. $codeOwnerConsoleOutput"
+    return $null
+  }
+  $codeOwnersMatches = $Matches[1] 
+  $codeOwnerJson = $codeOwnersMatches | ConvertFrom-Json
+  if ($codeOwnersJson) {
+    Write-Host "There is something wrong with parsing logic. Check output: $codeOwnerConsoleOutput"
+    return $null
+  }
+  
+  $codeOwners = $codeOwnerJson.Owners -join ","
+  Write-Host "##vso[task.setvariable variable=$VsoVariable;]$codeOwners"
+  return $codeOwners
 }
-else {
-  Write-Host "Unable to match path $target in CODEOWNERS file located at $codeOwnersLocation."
-  Write-Host ($ownedFolders | ConvertTo-Json)
-  return ""
+
+if (!(Test-Path "$ToolPath/retrieve-code-owners.exe")) {
```

Comment:
Will this be a .exe? 

We should move this check into the installRetrieveCodeOwnersTool function and always call it.

Reply:
+1 on the cross-platform tool check and not `.exe`